spyder/plugins/updatemanager/widgets/update.py

```diff
@@ -422,13 +431,10 @@ def start_install(self):
         shutil.copy2(script_path, tmpscript_path)
 
         # Sub command
-        sub_cmd = [tmpscript_path, '-p', sys.prefix]
-        if osp.exists(self.installer_path):
-            # Run downloaded installer
-            sub_cmd.extend(['-i', self.installer_path])
-        elif self.latest_release is not None:
+        sub_cmd = [tmpscript_path, '-i', self.installer_path]
+        if self.installer_path.endswith('.lock'):
             # Update with conda
-            sub_cmd.extend(['-c', find_conda(), '-v', self.latest_release])
+            sub_cmd.extend(['-c', find_conda(), '-p', sys.prefix])
```

Comment:
A couple of questions about this:

* So `conda` can perform the update with a conda-lock file? That's really cool!
* Do mamba or micromamba come with the installer? Just asking because I guess that would make the update process faster (although if every package is pinned to a specific version, conda wouldn't need to solve their dependencies, right?).

Reply:
We include mamba in the root environment of the installer; this will be the executable that Spyder's `find_conda()` will return for our installer based applications. I'll explain the implications of conda-lock, which provides 3 different file flavors.

1. **Standard lock file**. This is the universal and most comprehensive lock file and can be used to "render" the following two lock files types. It uses the extension `.yml`. However, _only_ `conda-lock` can be used with this file, i.e. `conda-lock install conda-lock.yml`. This operation does _not_ run the solver. This cannot be used to _update_ an environment, only create one. This will completely remove an existing environment and replace it. For this reason we cannot use this file; when clobbering the existing `spyder-runtime` environment, we would lose the `Menu/conda-based-app` file created by the installer that signals `menuinst` that this is an installer-based environment (as opposed to a standard conda environment).
2. **Explicit lock file**. This is just as explicit as the standard lock file, but is specific to a platform, and has the extension `.lock`. The file will include the platform in the name (e.g. `conda-osx-x86_64.lock`). Any name template _must_ include `{platform}` for this reason. This can be used with `conda-lock`, but will also _replace_ an existing environment; we cannot do that. This file can _also_ be used with `mamba`/`conda` and can be used to _update_ the environment, e.g. `mamba update -n spyder-runtime --file conda-osx-x86_64.lock`. This is what we must do.
3. **Environment lock file**. This is identical to an `environment.yml` file, specific to a platform, with all packages pinned. This has the extension `.lock.yml` but otherwise has the same name pattern as the explicit lock file. This will also clobber an existing environment when used with `conda-lock`, so we cannot do that. This can be used with `mamba`/`conda` as well, since it is just a standard environment spec file, however, the solver _will_ be invoked in this case. For this reason the explicit lock file is the best choice for our application.